src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,15 +1,106 @@
-import { Component } from 'react';
+import { ChangeEvent, FormEvent, PureComponent } from 'react';
+import { Movie } from '../../types/types';
+import './NewMovie.scss';
 
-type Props = {};
-type State = {};
+type Props = {
+  onAdd: (movie: Movie) => void;
+};
 
-export class NewMovie extends Component<Props, State> {
-  state: State = {};
+type State = Movie;
+
+export class NewMovie extends PureComponent<Props, State> {
+  state: State = {
+    title: '',
+    description: '',
+    imgUrl: '',
+    imdbUrl: '',
+    imdbId: '',
+  };
+
+  resetForm = () => {
+    this.setState({
+      title: '',
+      description: '',
+      imgUrl: '',
+      imdbUrl: '',
+      imdbId: '',
+    });
+  };
+
+  handleChange = (e: ChangeEvent<HTMLInputElement>) => {
```

Comment:
Не скорочуй назви змінних
```suggestion
  handleChange = (event: ChangeEvent<HTMLInputElement>) => {
```